### PR TITLE
cmd: fix a bunch of linter warnings from GoLand

### PR DESCRIPTION
**Pull request by derekparker**
Also fixes a potential nil pointer deref error in a test.

### CODE REVIEW

File: cmd/dlv/dlv_test.go [hunk 1]
```diff
@@ -180,17 +180,17 @@ func testOutput(t *testing.T, dlvbin, output string, delveCmds []string) (stdout
 
 	// ignore "dlv debug" command error, it returns
 	// errors even after successful debug session.
-	cmd.Wait()
+	_ = cmd.Wait()
```
Comment (aarzilli) [inline cmd/dlv/dlv_test.go:183]: Let's not do this, it's everywhere and it looks ugly.

File: cmd/dlv/dlv_test.go [hunk 2]
```diff
@@ -180,17 +180,17 @@ func testOutput(t *testing.T, dlvbin, output string, delveCmds []string) (stdout
...
 	_, err = os.Stat(debugbin)
 	if err == nil {
 		// Sometimes delve on Windows can't remove the built binary before
 		// exiting and gets an "Access is denied" error when trying.
-		// See: https://travis-ci.com/go-delve/delve/jobs/296325131)
+		// See: https://travis-ci.com/go-delve/delve/jobs/296325131.
 		// We have added a delay to gobuild.Remove, but to avoid any test
 		// flakiness, we guard against this failure here as well.
-		if runtime.GOOS != "windows" || !strings.Contains(err.Error(), "Access is denied") {
+		if runtime.GOOS != "windows" {
```
Comment (aarzilli) [inline cmd/dlv/dlv_test.go:193]: But we do want to complain for errors other than access is denied.

File: cmd/dlv/main.go [hunk 3]
```diff
@@ -15,11 +15,16 @@ func main() {
 	if Build != "" {
 		version.DelveVersion.Build = Build
 	}
+
 	const cgoCflagsEnv = "CGO_CFLAGS"
 	if os.Getenv(cgoCflagsEnv) == "" {
-		os.Setenv(cgoCflagsEnv, "-O0 -g")
+		err := os.Setenv(cgoCflagsEnv, "-O0 -g")
+		if err != nil {
```
Comment (aarzilli) [inline cmd/dlv/main.go:22]: os.Setenv can't fail except for a malformed key.

File: cmd/dlv/dlv_test.go (hunk 1 above)
Comment (derekparker) [reply cmd/dlv/dlv_test.go]: Hah, agreed, automated change. Removed.

File: cmd/dlv/dlv_test.go (hunk 2 above)
Comment (derekparker) [reply cmd/dlv/dlv_test.go:193]: The enclosing branch checks that `err == nil`.

Comment (aarzilli) [reply cmd/dlv/dlv_test.go:193]: Ah ok, you are right.

File: cmd/dlv/main.go (hunk 3 above)
Comment (derekparker) [reply cmd/dlv/main.go]: True, reverted.

**Review by aarzilli — APPROVED** [review]

**Pull request merged**